core/src/main/kotlin/org/evomaster/core/problem/rest/RestPath.kt

```diff
@@ -148,6 +149,30 @@ class RestPath(path: String) {
         return (0 until elements.size).none { this.elements[it] != other.elements[it] }
     }
 
+    /**
+     * @return whether this is sibling of the [other]
+     *
+     * eg, we consider
+     * /root/{rootName}/foo/{fooName}/bar/{barName}
+     * /root/{rootName}/foo/{fooName}/bar are sibling
+     *
+     * note that in some cases,
+     *
+     * GET /root/{rootName}/foo/{fooName}/bar
+     * might need its sibling (eg, /root/{rootName}/foo/{fooName}/bar/{barName})
+     * to prepare resources
+     */
+    fun isSibling(other: RestPath): Boolean {
```

Comment:
i m not sure `sibling` is the right word here. sibling would mean something like "at the same level", whereas here it looks more a parent-child relations

Reply:
here, we consider that those two paths can prepare resources for each other, then use sibling 

POST /root/{rootName}/foo/{fooName}/bar/{barName}
GET /root/{rootName}/foo/{fooName}/bar

POST /root/{rootName}/foo/{fooName}/bar
GET /root/{rootName}/foo/{fooName}/bar/{barName}

the name of the method is modified and add more docs for clarifying this.